### DISCUSSION TITLE
NextJS + Strapi - Custom layout and dynamic values

### DISCUSSION THREAD

**Discussion by danestves** [tw-in-js/twind#150] [Show and tell]
As anyone in the industry of making a lot of websites for clients sometimes the clients wants to use alternatives like:

- https://github.com/emotion-js/emotion
- https://github.com/styled-components/styled-components

But this packages has a high bundle size, with `twind` the thing is different. We always use Tailwind for all of our projects but sometimes the clients want a custom fontSize, color, etc that comes from Strapi (it's our backend by default) and now in one of our biggest project we are changing from emotion-js to use `twind` for all the application

![image](https://user-images.githubusercontent.com/31737273/111809105-00d9f000-88ab-11eb-82b7-9b66db29f65f.png)

Here is a difference of what is looks like one of the components with styled vs `twind`

![image](https://user-images.githubusercontent.com/31737273/111809178-1a7b3780-88ab-11eb-953c-9be7ba6e2687.png)

And in the browser it's even more magical the way that is working 😍 we are waiting for the official release of the dynamic styles features to pass all the application to use only `twind` and leverage bundle size and also improve the development time

**Comment by danielweck** [comment]
Great :)

Tip: I was using `clsx` too (super little package!) ... but now I use `tw` for all my needs, so I removed the `clsx` dependency from my project. The `tw` function is very versatile.

**Comment by lukejacksonn** [comment]
This is awesome to see man! Congrats 🥳 we have also transitioned from Tailwind to Twind for a client recently so that we can accommodate composition more readily (using `apply` and `twind/style`).

One thing I notice here (maybe you have already figured this out) is that Twind has `clsx` style combinatory logic built in too. So for example:

```js
// Template literals
tw`bg-gray-200 rounded`
//=> bg-gray-200 rounded
tw`bg-gray-200 ${false && 'rounded'}`
//=> bg-gray-200

// Strings
tw('bg-gray-200', true && 'rounded', 'underline')
//=> bg-gray-200 rounded underline

// Object syntax
tw({ 'bg-gray-200': true, rounded: false, underline: isTrue() })
//=> bg-gray-200 underline
tw({ 'bg-gray-200': true }, { rounded: false }, null, { underline: true })
//=> bg-gray-200 underline
tw({ hover: ['bg-red-500', 'p-3'] }, 'm-1')
// => hover:bg-red-500 hover:p-3 m-1

// Arrays
tw(['bg-gray-200', 0, false, 'rounded'])
//=> bg-gray-200 rounded
tw(['bg-gray-200'], ['', 0, false, 'rounded'], [['underline']])
//=> bg-gray-200 rounded underline

// Variadic
tw('bg-gray-200', [
  1 && 'rounded',
  { underline: false, 'text-black': null },
  ['text-lg', ['shadow-lg']],
])
//=> bg-gray-200 rounded text-lg shadow-lg
```

Taken from the docs here https://twind.dev/docs/modules/twind.html#tw-function

**Reply by danielweck** [reply]
That’s what I love about Twind: it promotes the Tailwind CSS design pattern, but the API is very flexible so there is an escape hatch for pretty much any imaginable custom styling requirement (for example to temporarily work around selector precedence issues, or to permanently setup global rules, etc.)

**Reply by danestves** [reply]
OMG let me try it!

**Reply by danestves** [reply]
![image](https://user-images.githubusercontent.com/31737273/111834532-1d395500-88ca-11eb-8717-9896b152db82.png)

@sastan maybe i'm missing something?

**Reply by danestves** [reply]
![image](https://user-images.githubusercontent.com/31737273/111835303-2bd43c00-88cb-11eb-87d5-7bde349902dd.png)
 
@sastan as i see is happening on setup too

**Reply by sastan** [reply]
I don't know. There must be some other problem. See this example: https://codesandbox.io/s/twind-nextjs-forked-0qjdh?file=/pages/index.js

**Reply by danestves** [reply]
layer 8 error sorry

**Comment by danestves** [comment]
@sastan here is the site on production with `twind` https://seeed.us/ 🎉
